app/src/main/java/com/pajato/android/gamechat/main/MainService.java

```diff
@@ -17,150 +17,54 @@
- * the service has been informed of the current User (via an account change event).
- *
- * @author Paul Michael Reilly on 2/25/17
+ * @author Paul Michael Reilly on 3/15/17
  */
-public class MainService extends Service {
-
-    // Public enums
-
-    /** The connand types. */
-    public enum CommandType {
-        none, onDestroy, onPause, onResume,
-    }
+public class MainService extends IntentService {
 
     // Public class constants.
 
-    /** The command key for the intent extra providing the command. */
-    public static final String COMMAND_KEY = "commandKey";
+    /** Provide an intent key the the room push value. */
+    public static final String ROOM_KEY = "roomKey";
 
     // Private class constants.
 
     /** The logcat tag. */
-    private static final String TAG = "MainService";
-
-    // Public instance methods.
-
-    /** Implement the Service interface by providing a null operation (no binding is done). */
-    @Override public IBinder onBind(final Intent intent) {
-        logEvent("bind", "not used");
-        return null;
-    }
-
-    /** Implement the Service interface by initializing the service on startup. */
-    @Override public void onCreate() {
-        logEvent("create", "Creating the main service");
-        init();
-    }
-
-    /** Implement the Service interface by handling the death of the service. */
-    @Override public void onDestroy() {
-        logEvent("destroy", "Destroying the main service");
-        // TODO: figure this out
-    }
-
-    /** Implement the Service interface to handle a command. */
-    @Override public int onStartCommand(final Intent intent, final int flags, final int startId) {
-        // Log the command and ensure that it is valid.   dispatch to process it.
-        String name = intent.getStringExtra(COMMAND_KEY);
-        String format = "onStartCommand: executing command with name {%s}, flags {%x}, id {%d}";
-        logEvent("command", String.format(Locale.US, format, name, flags, startId));
-        CommandType type = getCommandType(name);
-        if (type == null)
-            return handleInvalidCommand(name);
+    private static final String TAG = MainService.class.getSimpleName();
 
-        // Determine if the app has been restarted or is a mode change.
-        if (type == onDestroy)
-            // The app is restarting.  Update notifications for all registered Users on this device.
-            // for each User: log in, set up listeners; start an alarm; listen for
-            // message/experience changes (cache the most recently seen)
-            return handleNotificationUpdates();
+    // Public constructor
 
-        // Persist the mode change via shared preferences.
-        saveMode(type);
-        return Service.START_NOT_STICKY;
+    /** Build an instance to name the worker thread. */
+    public MainService() {
+        super("NotificationIntentService");
     }
 
-    // Private instance methods.
+    // Public instance methods.
```

Comment:
This is changed to 'Public instance methods' but the following method is 'protected'.